app/src/main/java/openfoodfacts/github/scrachx/openfood/views/LoadTaxonomiesService.java

```diff
@@ -56,45 +57,121 @@ private void doTask(Intent intent) {
+                    //.doOnSubscribe(d -> view.showLoading())
+                    .subscribe(() -> {
+                        //view.hideLoading(false);
+                      //   builder.setContentText(getString(R.string.txtLoaded));
+                     //    builder.setOngoing(false);
+                     //    notificationManager.notify(17, builder.build());
+                    }, e -> {
+                       //  e.printStackTrace();
+                        //view.hideLoading(true);
+                       // builder.setContentText(getString(R.string.txtConnectionError))
+                        //        .setOngoing(false);
+                        // notificationManager.notify(17, builder.build());
+                    });
+        } else if (BuildConfig.FLAVOR.equals("obf")) {
+            Single.zip(
+                    productRepository.getLabels(true),
+                    productRepository.getTags(true),
+                    productRepository.getIngredients(false), //TODO : have a test on last-modified-date of ingredients.json before download it. Then pass the parameter to true
+                    productRepository.getCountries(true),
+                    productRepository.getAdditives(true),
+                    productRepository.getCategories(true), (labels, tags, ingredients, countries, additives, categories) -> {
+                        Completable.merge(
+                                Arrays.asList(
+                                        Completable.fromAction(() -> productRepository.saveLabels(labels)),
+                                        Completable.fromAction(() -> productRepository.saveTags(tags)),
+                                        Completable.fromAction(() -> productRepository.saveIngredients(ingredients)),
+                                        Completable.fromAction(() -> productRepository.saveCountries(countries)),
+                                        Completable.fromAction(() -> productRepository.saveAdditives(additives)),
+                                        Completable.fromAction(() -> productRepository.saveCategories(categories))
+                                )
+                        ).subscribeOn(Schedulers.computation())
+                                .subscribe(() -> {
+                                    settings.edit().putLong(Utils.LAST_REFRESH_DATE, System.currentTimeMillis()).apply();
+                                }, Throwable::printStackTrace);
+
+                        return true;
+                    })
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .toCompletable()
+                    //.doOnSubscribe(d -> view.showLoading())
+                    .subscribe(() -> {
+                        //view.hideLoading(false);
+                      //   builder.setContentText(getString(R.string.txtLoaded));
+                     //    builder.setOngoing(false);
```

Comment:
same review ( to clean old code...)

Reply:
Same comment. waiting for your same answer...

Reply:
Commented code cleanup